tests/rbd_mirror/test_rbd_mirror_snapshot.py

```diff
@@ -0,0 +1,92 @@
+    """
+    --> Configures RBD Mirroring on cephadm
+    --> Creates Pool Image and enables snapshot based Mirroring on the pool
+    --> Runs IO using rbd bench
+    --> Creates images on the pool and verifies snapshots are created for each image
+    --> Cleanup
+    Args:
+        **kw:
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+    """
+    try:
+        log.info("Starting snapshot RBD mirroring test case")
+        config = kw.get("config")
+        mirror1, mirror2 = [
+            rbdmirror.RbdMirror(cluster, config)
+            for cluster in kw.get("ceph_cluster_dict").values()
+        ]
+        poolname = mirror1.random_string() + "_tier_1_rbd_mirror_pool"
+        imagename = mirror1.random_string() + "_tier_1_rbd_mirror_image"
+        imagespec = poolname + "/" + imagename
+
+        mirror1.create_pool(poolname=poolname)
+        mirror2.create_pool(poolname=poolname)
+        mirror1.create_image(imagespec=imagespec, size=config.get("imagesize"))
+        mirror1.config_mirror(mirror2, poolname=poolname, mode="image")
+        mirror1.enable_mirror_image(poolname, imagename, "snapshot")
+        mirror2.wait_for_status(poolname=poolname, images_pattern=1)
+        mirror1.benchwrite(imagespec=imagespec, io=config.get("io-total"))
+        mirror1.wait_for_status(imagespec=imagespec, state_pattern="up+stopped")
+        mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
+
+        # check if snapshots are created for image created above
+        snapshot_schedule_level = config.get("snapshot_schedule_level")
+        if not snapshot_schedule_level or snapshot_schedule_level == "cluster":
+            mirror1.mirror_snapshot_schedule_add()
+        elif snapshot_schedule_level == "pool":
+            mirror1.mirror_snapshot_schedule_add(poolname=poolname)
+        else:
+            mirror1.mirror_snapshot_schedule_add(poolname=poolname, imagename=imagename)
+        mirror1.verify_snapshot_schedule(imagespec)
+        mirror1.mirror_snapshot_schedule_list(poolname=poolname, imagename=imagename)
+        mirror1.mirror_snapshot_schedule_status(poolname=poolname, imagename=imagename)
+
+        # create one more image in the pool and check if snapshots are created
+        imagename_2 = mirror1.random_string() + "_tier_1_rbd_mirror_image"
+        imagespec_2 = poolname + "/" + imagename_2
+        mirror1.create_image(imagespec=imagespec_2, size=config.get("imagesize"))
+        mirror1.enable_mirror_image(poolname, imagename_2, "snapshot")
+        if snapshot_schedule_level == "image":
+            mirror1.mirror_snapshot_schedule_add(
+                poolname=poolname, imagename=imagename_2
+            )
+        mirror1.verify_snapshot_schedule(imagespec_2)
+        mirror1.mirror_snapshot_schedule_list(poolname=poolname, imagename=imagename_2)
+        mirror1.mirror_snapshot_schedule_status(
+            poolname=poolname, imagename=imagename_2
+        )
+
```

Comment:
can we have short doc for snapshot removal group steps.
it would be better if we have verification for snapshot schedule removal.

Reply:
@rkhambad Sure, I will add short doc describing how scheduling is removed here. is it okay if i add verification for snapshot schedule removal as a different task and take it up later?